### PR TITLE
devops/1370 - Upgrade dependencies and address npm audit issues

### LINKED ISSUE #1370: Upgrade deps and npm audit
- [x] upgrade to `gulp@latest` and supporting packages
- [x] upgrade `ava`
- [x] upgrade `eslint` and supporting packages
- [x] adress `npm audit` concerns

### PR DESCRIPTION
**Pull request by n8rzz**
Resolves #1370 

Upgrades dependencies and addresses `npm audit` issues

### CODE REVIEW

File: test/aircraft/AircraftModel.spec.js [hunk 1]
```diff
@@ -254,84 +258,91 @@ ava('.isEstablishedOnGlidepath() returns true when an acceptable distance below
...
+
+    isEstablishedOnCourseStub.restore();
+    distanceToPositionStub.restore();
 });
 
-ava('._calculateArrivalRunwayModelGlideslopeAltitude() returns arrival runway\'s glideslope altitude abeam the specified position', (t) => {
-    const model = new AircraftModel(ARRIVAL_AIRCRAFT_INIT_PROPS_MOCK);
-    const expectedResult = 3994.129742601768;
+// TODO: this test is suddenly failing entirely
+// ava('._calculateArrivalRunwayModelGlideslopeAltitude() returns arrival runway\'s glideslope altitude abeam the specified position', (t) => {
```
Comment (erikquinn) [inline test/aircraft/AircraftModel.spec.js]: Any idea why?

File: test/aircraft/Pilot/Pilot.spec.js [hunk 2]
```diff
@@ -855,7 +855,7 @@ ava('.initiateHoldingPattern() returns correct readback when hold implemented su
     t.deepEqual(result, expectedResult);
 });
 
-ava('.maintainAltitude() returns early responding that they are unable to maintain the requested altitude', (t) => {
+ava.skip('.maintainAltitude() returns early responding that they are unable to maintain the requested altitude', (t) => {
```
Comment (erikquinn) [inline test/aircraft/Pilot/Pilot.spec.js]: Should this be skipped?

File: test/parsers/aircraftCommandParser/AircraftCommandParser.spec.js [hunk 3]
```diff
@@ -33,9 +33,15 @@ ava('throws when called with an invalid command', (t) => {
...
 ava('throws when called with invalid arguments', (t) => {
+    const expectedResult = 'Invalid argument length. Expected exactly zero arguments';
     const commandStringMock = buildCommandString(TAKEOFF_MOCK, 'threeve');
 
-    t.throws(() => new AircraftCommandParser(commandStringMock));
+    try {
+        // eslint-disable-next-line no-unused-vars
+        const model = new AircraftCommandParser(commandStringMock);
+    } catch (e) {
+        t.true(e === expectedResult);
```
Comment (erikquinn) [inline test/parsers/aircraftCommandParser/AircraftCommandParser.spec.js]: Hm... that's a good idea. Don't know why I never did that before 🤔

File: test/aircraft/Pilot/Pilot.spec.js (hunk 2 above)
Comment (n8rzz) [reply test/aircraft/Pilot/Pilot.spec.js]: Yeah, both of these tests are failing after the upgrades.  If I remember correctly, it's just a precision thing due to a newer version of node, ava, etc and I haven't had time to go back and address it.

File: test/aircraft/AircraftModel.spec.js (hunk 1 above)
Comment (n8rzz) [reply test/aircraft/AircraftModel.spec.js]: No idea why, but the difference is substantial:

Screen Shot 2019-05-21 at 8 28 41 PM

File: test/aircraft/Pilot/Pilot.spec.js (hunk 2 above)
Comment (n8rzz) [reply test/aircraft/Pilot/Pilot.spec.js]: this is passing now, just removed the `.skip()`

File: test/aircraft/AircraftModel.spec.js (hunk 1 above)
Comment (erikquinn) [reply test/aircraft/AircraftModel.spec.js]: fixed this.

**Pull request merged**